fdbserver/BulkLoadUtil.actor.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * BulkLoadUtils.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbclient/ClientKnobs.h"
+#include "fdbserver/BulkLoadUtil.actor.h"
+#include "fdbserver/RocksDBCheckpointUtils.actor.h"
+#include "fdbserver/StorageMetrics.actor.h"
+#include <fmt/format.h>
+#include "flow/actorcompiler.h" // has to be last include
+
+std::string generateRandomBulkLoadDataFileName() {
+	return deterministicRandom()->randomUniqueID().toString() + "-data.sst";
+}
+
+std::string generateRandomBulkLoadBytesSampleFileName() {
+	return deterministicRandom()->randomUniqueID().toString() + "-bytesample.sst";
+}
+
+ACTOR Future<Optional<BulkLoadState>> getBulkLoadState(Database cx, UID dataMoveId, UID logId) {
+	// TODO(Zhe): check if bulk load task is outdated
+	loop {
+		state Transaction tr(cx);
+		try {
+			Optional<Value> val = wait(tr.get(dataMoveKeyFor(dataMoveId)));
+			if (!val.present()) {
+				TraceEvent(SevWarn, "SSBulkLoadDataMoveIdNotExist", logId).detail("DataMoveID", dataMoveId);
+				return Optional<BulkLoadState>(); // TODO(Zhe): is this case not expected?
+			}
+			DataMoveMetaData dataMoveMetaData = decodeDataMoveValue(val.get());
```

Comment:
Why not

			return decodeDataMoveValue(val.get());

instead of

			DataMoveMetaData dataMoveMetaData = decodeDataMoveValue(val.get());
			return dataMoveMetaData.bulkLoadState;

Reply:
This function is returning BulkLoadState instead of DataMoveMetaData, since the function purpose is to get the bulk load state given the data move id.